api/v1alpha4/clusterclass_webhook.go

```diff
@@ -92,7 +96,7 @@ func (in *ClusterClass) validate(old *ClusterClass) error {
 	var allErrs field.ErrorList
 
 	// ensure all the references are within the same namespace
-	if in.Spec.Infrastructure.Ref != nil && in.Spec.Infrastructure.Ref.Namespace != in.Namespace {
```

Comment:
We dont need a nil check here as the openapischema validations throw an error even before we get here since this field is non-optional.

Reply:
OpenAPISchema validations are executed both and client- and server-side, right? Just because we had a case where we saw that they are at least execute on client-side.

Reply:
I am not sure if OpenAPISchema validations happen on the client-side, but i believe they happen on the server-side.

Reply:
Just to clarify client-side == kubectl. Our case was that we had some validation logic via OpenAPI and some defaulting in a webhook. Because the OpenAPI validation was executed in kubectl before defaulting it didn't work as intended. But I think you're right it's probably done twice.